test/nodejs/route.js

```diff
@@ -322,6 +322,12 @@ test('route: routes Monaco with options', function(assert) {
         assert.ok(first.routes[0].legs[0]);
         assert.ok(first.routes[0].legs.every(l => { return l.steps.length > 0; }), 'every leg has steps');
         assert.ok(first.routes[0].legs.every(l => { return l.annotation;}), 'every leg has annotations');
+        assert.ok(first.routes[0].legs.every(l => { return l.annotation.distance;}), 'every leg has annotations for distance');
+        assert.ok(first.routes[0].legs.every(l => { return l.annotation.duration;}), 'every leg has annotations for durations');
+        assert.ok(first.routes[0].legs.every(l => { return l.annotation.nodes;}), 'every leg has annotations for nodes');
+        assert.ok(first.routes[0].legs.every(l => { return l.annotation.weight; }), 'every leg has annotations for weight');
+        assert.ok(first.routes[0].legs.every(l => { return l.annotation.datasources; }), 'every leg has annotations for datasources');
+        assert.ok(first.routes[0].legs.every(l => { return l.annotation.speed; }), 'every leg has annotations for speed');
```

Comment:
Btw problem was happening only for `speed`, because we have these checks here which it seems were added for backward compatibility, but I am not sure if we should use `parameters.annotations_type`(but not `requested_annotations`) when checking if speed annotations were requested.
https://github.com/Project-OSRM/osrm-backend/blob/5e5f1f4add6d86b097569545b670de513ebc6b04/include/engine/api/route_api.hpp#L770
<img width="1267" alt="Screenshot 2022-10-18 at 18 35 51" src="https://user-images.githubusercontent.com/266271/196490626-0bf46f2d-49a4-4883-a147-9b1ca72f268a.png">

Reply:
Yep, looks like it.